File: Count Distinct Elements in Every Window.c
#include <stdio.h>
#include <stdlib.h>

void countDistinct(int arr[], int n, int k) {
    for(int i = 0; i <= n - k; i++) {
        int count = 0;

        for(int j = i; j < i + k; j++) {
            int isDistinct = 1;

            for(int p = i; p < j; p++) {
                if(arr[p] == arr[j]) {
                    isDistinct = 0;
                    break;
                }
            }

            if(isDistinct) count++;
        }

        printf("%d ", count);
    }
}

File: Count Distinct Elements in Every Window.py
def count_distinct(arr, k):
    n = len(arr)
    result = []

    freq = {}

    # first window
    for i in range(k):
        freq[arr[i]] = freq.get(arr[i], 0) + 1

    result.append(len(freq))

    for i in range(k, n):
        # remove previous
        freq[arr[i - k]] -= 1
        if freq[arr[i - k]] == 0:
            del freq[arr[i - k]]

        # add new
        freq[arr[i]] = freq.get(arr[i], 0) + 1

        result.append(len(freq))

    return result
